backend/controller/dal/encryption.go

```diff
@@ -84,5 +88,80 @@ func (d *DAL) EnsureKey(ctx context.Context, generateKey func() ([]byte, error))
 	}
 
 	logger.Debugf("Encryption key found, using it")
-	return encryptedKey, nil
+
+	return row.Key, nil
+}
+
+const verification = "FTL - Towards a 𝝺-calculus for large-scale systems"
+
+func (d *DAL) verifyEncryptor(ctx context.Context) (err error) {
+	var tx *Tx
+	tx, err = d.Begin(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to begin transaction: %w", err)
+	}
+	defer tx.CommitOrRollback(ctx, &err)
+
+	var row sql.GetOnlyEncryptionKeyRow
+	row, err = tx.db.GetOnlyEncryptionKey(ctx)
+	if err != nil {
+		if dal.IsNotFound(err) {
+			// No encryption key found, probably using noop.
+			return nil
+		}
+		return fmt.Errorf("failed to get encryption row from the db: %w", err)
+	}
+
+	up := false
+	up, err = verifySubkey(d.encryptor, &row.VerifyTimeline)
+	if err != nil {
+		return fmt.Errorf("failed to verify timeline subkey: %w", err)
+	}
+	needsUpdate := up
+
+	up, err = verifySubkey(d.encryptor, &row.VerifyAsync)
+	if err != nil {
+		return fmt.Errorf("failed to verify async subkey: %w", err)
+	}
+	needsUpdate = needsUpdate || up
+
+	if !needsUpdate {
+		return nil
+	}
+
+	if !row.VerifyTimeline.Ok() || !row.VerifyAsync.Ok() {
+		panic("should be unreachable. verifySubkey should have set the subkey")
+	}
+
+	err = tx.db.UpdateEncryptionVerification(ctx, row.VerifyTimeline, row.VerifyAsync)
+	if err != nil {
+		return fmt.Errorf("failed to update encryption verification: %w", err)
+	}
+
+	return nil
+}
+
+// verifySubkey checks if the subkey is set and if not, sets it to a verification string.
+// It returns true if the subkey was set.
+func verifySubkey[SK encryption.SubKey](encryptor encryption.DataEncryptor, encrypted *optional.Option[encryption.EncryptedColumn[SK]]) (bool, error) {
```

Comment:
This is a bit of a nit, but passing around pointers to Option feels kinda wrong to me. Why not just return the resulting key?

Reply:
Good point. It was just the way I wrote it initially and tunnel visioned it. Will fix!